sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ReadmeSamples.java

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
@alzimmermsft  - we only have code snippets in samples directory, and I see from your PR, that we are avoiding using code snippet links, and directly embedding them. So why do we need this class and folder going forward ?

Reply:
So, the class could be renamed to whatever we liked and with additional configuration changes it could be move to another location such as tests. But, for now it's following a pattern where the `README.md` codesnippets are maintained and compiled in a `/src/samples/.../ReadmeSamples.java` and the `maven-codesnippet-plugin` we use (and maintain) will pull the `// BEGIN: <alias>` `// END: <alias>` code blocks and inject them into the `java <alias>` `README.md` code fences.

As an example in the README if it had:

_Ticks in the code fence are wrapped in ( ) to prevent it from closing early_

```
(```)java myCoolSample
(```)
```

And in my samples I had:

```
public void myCoolSampleCode() {
    // BEGIN: myCoolSample
    int onePlusOneEqualsTwo = 1 + 1;
    assert onePlusOneEqualsTwo == 2;
    // END: myCoolSample
}
```

When the maven-codesnippet-plugin is ran it'll copy the sample code into the README permanently, so any time there is a refactoring or code change made in the sample code it'll automatically propagate to the README (or Javadocs as it does those as well) on build.

Reply:
Sounds good, thanks for the explanation.